runtime/src/iree/base/status.h

```diff
@@ -498,6 +500,13 @@ IREE_API_EXPORT bool iree_status_format(iree_status_t status,
                                         char* buffer,
                                         iree_host_size_t* out_buffer_length);
 
+// Converts the status to an allocated string value using the given allocator.
+// The caller must free the buffer with |allocator|.
```

Comment:
This will be a head scratcher later and would like to avoid copy-pasta of the pattern going further. Can you add a comment that passing by reference avoids a circular dependency for these low level parts and should not be done elsewhere?

Reply:
Will do!

Reply:
Added comment + usage example here: https://github.com/iree-org/iree/commit/31162eba03ef4c6b3ae8a52540b1ea85b455ce57